### ISSUE TITLE
Handshake failure on packet loss

### ISSUE THREAD

**Issue by huitema** [private-octopus/picoquic#775]
The log of a connection test shows:
```
...
4c2af2ba41bece1c: Receiving 220 bytes from 193.167.100.100:443 at T=3.035412 (59c94f499ee8f)
4c2af2ba41bece1c: Receiving packet type: 2 (initial), S0, Version ff000018,
4c2af2ba41bece1c:     <38ed74018c065f3f>, <89f4425ec293c95b>, Seq: 2, pl: 194
4c2af2ba41bece1c:     Token length: 0
4c2af2ba41bece1c:     Decrypted 174 bytes
4c2af2ba41bece1c:     Crypto HS frame, offset 0, length 161: 0200009d030320fc...
4c2af2ba41bece1c:     ACK_1WD 5b943, (nb=0), 0-2

4c2af2ba41bece1c: Sending packet type: 2 (initial), S0, Version ff000018,
4c2af2ba41bece1c:     <89f4425ec293c95b>, <38ed74018c065f3f>, Seq: 3, pl: 25
4c2af2ba41bece1c:     Token length: 0
4c2af2ba41bece1c:     Prepared 5 bytes
4c2af2ba41bece1c:     ACK (nb=0), 2

4c2af2ba41bece1c: Sending packet type: 5 (0rtt protected), S0, Version ff000018,
4c2af2ba41bece1c:     <4c2af2ba41bece1c>, <38ed74018c065f3f>, Seq: 1, pl: 1176
4c2af2ba41bece1c:     Prepared 1156 bytes
4c2af2ba41bece1c:     padding, 1156 bytes

4c2af2ba41bece1c: Sending 1252 bytes to 193.167.100.100:443 at T=3.035607 (59c94f499ef52)
```
And then nothing. At this point, the client does not repeat any packet because only an ACK is outstanding, and the server does not repeat any packet either because the connection is not validated. There are two issues:

1) The server should not have sent an ACK in the Initial packet if the session did not progress;
2) The client should keep pinging the server if the handshake is not complete.

**Comment by huitema** [comment]
Seems that the error still happens, at least in some cases.

**Comment by huitema** [comment]
Appears fixed by PR #772
